chapters/ch55.md

```diff
@@ -1,78 +1,82 @@
 
 > 3. Attribute the error to component C.
 
---> _replace THIS LINE by your translation for the above line_
+3. Sau khi đã kiểm chứng nguyên nhân phát sinh lỗi không phải do A và B, ta có thể quy lỗi cho thành phần C. 
+
 
 > Let’s look at a more complex example:
 
---> _replace THIS LINE by your translation for the above line_
+Chúng ta hãy cùng tìm hiểu một ví dụ phức tạp hơn sau đây:
 
 ![img](../imgs/C55_02.png)
 
 > Your self-driving car uses this pipeline. How do you use error analysis by parts to decide which component(s) to focus on?
 
---> _replace THIS LINE by your translation for the above line_
+Xe hơi tự lái của bạn sử dụng hệ thống pipeline như trên. Bạn sẽ sử dụng kỹ thuật phân tích lỗi từng phần như thế nào để quyết định thành phần (hay nhóm thành phần) nào cần cải thiện?
 
 > You can map the three components to A, B, C as follows:<br/>
 > A: Detect cars<br/>
 > B: Detect pedestrians <br/>
 > C: Plan path for car
 
---> _replace THIS LINE by your translation for the above line_
+Bạn có thể gọi tên ba thành phần trong hệ thống là A, B, C tương ứng với các chức năng như sau:
+A: phát hiện xe hơi
+B: phát hiện người đi bộ
+C: hoạch định đường đi cho xe
 
 > Following the procedure described above, suppose you test out your car on a closed track and find a case where the car chooses a more jarring steering direction than a skilled driver would. In the self-driving world, such a case is usually called a ​**scenario​**. You would then:
 
---> _replace THIS LINE by your translation for the above line_
+Với hệ thống xe tự lái mô tả như trên, giả sử bạn kiểm tra xe của bạn trên một cung đường kín và xác định trường hợp nào xe chọn hướng bẻ lái giật nhiều hơn so với một người lái xe kinh nghiệm điều khiển. Trong lĩnh vực lái xe tự động, một trường hợp như thế thường được gọi là "tình huống". Bạn cần thực hiện:
 
 > 1. Try manually modifying A (detecting cars)’s output to be a "perfect" output (e.g., manually go in and tell it where the other cars are). Run the rest of the pipeline B, C as before, but allow C (plan path) to use A’s now perfect output. If the algorithm now plans a much better path for the car, then this shows that, if only A had given a better output, the overall algorithm’s output would have been better; Thus, you can attribute this error to component A. Otherwise, go on to Step 2.
 
---> _replace THIS LINE by your translation for the above line_
+1. Thử điều chỉnh thủ công kết quả đầu ra của thành phần A (phát hiện xe hơi) sao cho "hoàn hảo" (ví dụ, cho xe biết vị trí của những chiếc xe khác). Sau đó chạy thuật toán trên  pipeline gồm có B và C, sủ dụng đầu ra "hoàn hảo" của A đã được thao tác thủ công trước khi đưa vào thành phần C (hoạch định đường đi cho xe). Nếu thuật toán hoạch định đường đi cho xe tốt hơn, điều đó cho thấy rằng, kết quả cuối cùng của thuật toán tự lái sẽ được cải thiện khi và chi khi thành phần A cho kết quả đầu ra tốt hơn.  Như vậy, bạn có thể quy lỗi cho A. Ngược lại, ta tiếp tục bước 2. 
```

Comment:
* "Sau đó chạy thuật toán trên  pipeline gồm có B và C, sủ dụng đầu ra "hoàn hảo" của A đã được thao tác thủ công trước khi đưa vào thành phần C (hoạch định đường đi cho xe)."
-> "Sau đó tiếp tục chạy phần còn lại của B, C của pipeline, nhưng cho phép C (hoạch định đường đi) sử dụng đầu ra đã hoàn hảo của A."

*  "if only" tương tự chỗ này

* "Ngược lại, ta tiếp tục bước 2. " -> "Nếu không, ..."

Reply:
> "Sau đó chạy thuật toán trên pipeline gồm có B và C, sủ dụng đầu ra "hoàn hảo" của A đã được thao tác thủ công trước khi đưa vào thành phần C (hoạch định đường đi cho xe)."
> -> "Sau đó tiếp tục chạy phần còn lại của B, C của pipeline, nhưng cho phép C (hoạch định đường đi) sử dụng đầu ra đã hoàn hảo của A."

Mình góp ý nhỏ là ta có điều chỉnh chút xíu chỗ này "Sau đó tiếp tục chạy phần còn lại của B, C của pipeline" -->  "Sau đó tiếp tục chạy phần còn lại của pipeline gồm có B, C,"